broker-core/src/test/java/org/apache/qpid/server/security/encryption/AESGCMKeyFileEncrypterTest.java

```diff
@@ -0,0 +1,364 @@
+            final SettableFuture<SystemConfig<?>> configFuture = SettableFuture.create();
+            _systemLauncher = new SystemLauncher(new SystemLauncherListener.DefaultSystemLauncherListener()
+            {
+                @Override
+                public void onContainerResolve(final SystemConfig<?> systemConfig)
+                {
+                    configFuture.set(systemConfig);
+                }
+            });
+            _systemLauncher.startup(attributes);
+            final SystemConfig<?> systemConfig = configFuture.get(10, TimeUnit.SECONDS);
+            _broker = (Broker<?>) systemConfig.getContainer();
+
+            final Map<String, Object> authProviderAttributes = new HashMap<>();
+            authProviderAttributes.put(ConfiguredObject.NAME, "testAuthProvider");
+            authProviderAttributes.put(ConfiguredObject.TYPE, "Plain");
+
+            final AuthenticationProvider
+                    authProvider = _broker.createChild(AuthenticationProvider.class, authProviderAttributes);
+
+            Map<String, Object> userAttrs = new HashMap<>();
+            userAttrs.put(User.TYPE, "managed");
+            userAttrs.put(User.NAME, "guest");
+            userAttrs.put(User.PASSWORD, PLAINTEXT);
+            User user = (User) authProvider.createChild(User.class, userAttrs);
+        }finally{
+            _systemLauncher.shutdown();
+            FileUtils.deleteDirectory(_workDir.toFile().getAbsolutePath());
+            FileUtils.deleteDirectory(_configurationLocation.toFile().getAbsolutePath());
+        }
+    }
+
+    @Test
+    public void testRepeatedEncryptionsReturnDifferentValues() throws Exception
+    {
+        SecretKeySpec secretKey = createSecretKey();
+        AESGCMKeyFileEncrypter encrypter = new AESGCMKeyFileEncrypter(secretKey);
+
+        Set<String> encryptions = new HashSet<>();
+
+        int iterations = 100;
+
+        for (int i = 0; i < iterations; i++)
+        {
+            encryptions.add(encrypter.encrypt(PLAINTEXT));
+        }
+
+        assertEquals("Not all encryptions were distinct", (long) iterations, (long) encryptions.size());
+
+        for (String encrypted : encryptions)
+        {
+            assertEquals("Not all encryptions decrypt correctly", PLAINTEXT, encrypter.decrypt(encrypted));
+        }
+    }
+
+    @Test
+    public void testCreationFailsOnInvalidSecret() throws Exception
+    {
+        try
+        {
+            new AESGCMKeyFileEncrypter(null);
+            fail("An encrypter should not be creatable from a null key");
+        }
+        catch (NullPointerException e)
+        {
+            assertTrue("Unexpected exception message:" + e.getMessage(),
+                       e.getMessage().contains("A non null secret key must be supplied"));
+        }
+
+        try
+        {
+            PBEKeySpec keySpec = new PBEKeySpec(PLAIN_SECRET.toCharArray());
+            SecretKeyFactory factory = SecretKeyFactory.getInstance("PBEWithMD5AndDES");
```

Comment:
Does SecretKeyFactory "PBEWithMD5AndDES"exist in JVM

Reply:
yes.. It is being created!